src/auth/services/auth.service.ts

```diff
@@ -0,0 +1,55 @@
+import { Injectable } from '@nestjs/common';
+import { CustomersService } from 'src/customers/service/customers.service';
+import { Customer } from 'src/customers/domain/customer.domain';
+import { JwtService } from '@nestjs/jwt';
+import { compare } from 'bcrypt';
+
+@Injectable()
+export class AuthService {
+  constructor(
+    private readonly customersService: CustomersService,
+    private jwtService: JwtService,
+  ) {}
+
+  async validateCustomer(username: string, password: string) {
+    const customer =
+      await this.customersService.findCustomerByUsername(username);
+
+    if (customer && (await compare(password, customer.password))) {
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      const { password, ...customerData } = customer;
+      return customerData;
+    }
+    // console.log('password incorrect');
+    return null;
+  }
+
+  async logIn(customer: Customer) {
+    const valid = await this.validateCustomer(
+      customer.username,
+      customer.password,
+    );
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    const { password, ...customerData } = customer;
+    const payload = { sub: customer.id, role: valid.role };
+
+    return {
+      ...customerData,
+      accessToken: this.jwtService.sign(payload),
+      refreshToken: this.jwtService.sign(payload, { expiresIn: '7d' }),
+    };
+  }
+
+  async refrehToken(customer: Customer) {
```

Comment:
Typo: `refresh`, `access` below.

The refresh was not required, but I am curious if this works for you as you include both `id` and `role` in the `sub`. `Sub` should be a unique string and the role should be provided separated as you did inside the `logIn` function.